### PR TITLE
Artifacts support

### PR DESCRIPTION
**Pull request by kimpaller**

### CODE REVIEW

File: telemetry/gparser/parser.py [hunk 1]
```diff
@@ -18,13 +20,18 @@ def get_parser(url):
...
     }
 
     # find parser
     for sk, parser in parsers.items():
         if re.match(sk, url):
-            return parser(url)  
+            if isinstance(parser, list):
+                return [p(url) for p in parser]
+            else:
+                return parser(url)
```
Comment (tfcollins) [inline telemetry/gparser/parser.py]: The else is unnecessary. Should just be:
```python
if isinstance(parser, list):
    return [p(url) for p in parser]
return parser(url)
```
or maybe
```python
parser = parser if isinstance(parser, list) else [parser]
return [p(url) for p in parser]
```

File: telemetry/gparser/parser.py [hunk 2]
```diff
@@ -192,4 +207,81 @@ def __init__(self, url):
...
+            elif isinstance(self, PytestSkipped):
+                file_info = "pytest_skipped"
+            elif isinstance(self, PytestError):
+                file_info = "pytest_error"
+            target_board = file_name.replace('_','-')
+            target_board = remove_suffix(target_board,"-reports.xml")
+            artifact_info_type=file_info
+            return (file_name, file_info, target_board, artifact_info_type)
+        else:
+            raise Exception("Does not support non multilevel yet!")
```
Comment (tfcollins) [inline telemetry/gparser/parser.py]: else unnecessary

File: telemetry/gparser/parser.py (hunk 1 above)
Comment (kimpaller) [reply telemetry/gparser/parser.py]: https://github.com/sdgtt/telemetry/commit/84f663669ed98fb8e9fe4e5e7eea22438c726ebc

File: telemetry/gparser/parser.py (hunk 2 above)
Comment (kimpaller) [reply telemetry/gparser/parser.py]: https://github.com/sdgtt/telemetry/commit/84f663669ed98fb8e9fe4e5e7eea22438c726ebc

**Pull request merged**